### DISCUSSION TITLE
How to use libfdipp1?

### DISCUSSION THREAD

**Discussion by Enakers** [microsoft/vcpkg#40846] [Q&A]
Hi, new to vpckg and I cannot work out how to get libftdipp1 port to work, hopefully someone can point me in the right direction.

```
❯ pkg-config --version
2.1.1
```

```
❯ vcpkg --version
vcpkg package management program version 2024-08-01-fd884a0d390d12783076341bd43d77c3a6a
15658
```
```
❯ cmake --build build
[2/2] Linking CXX executable MyApp
FAILED: MyApp 
: && /usr/bin/c++   CMakeFiles/MyApp.dir/src/main.cpp.o -o MyApp  -lftdip
p1  vcpkg_installed/x64-linux/debug/lib/libftdi1.a  vcpkg_installed/x64-linux/debug/lib
/libusb-1.0.a  /usr/lib/libudev.so && :
/usr/bin/ld: cannot find -lftdipp1: No such file or directory
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

CMakeLists.txt
```
cmake_minimum_required(VERSION 3.30)

project(MyApp)

find_package(PkgConfig REQUIRED)

add_executable(MyApp src/main.cpp)

pkg_check_modules(FTDI REQUIRED IMPORTED_TARGET libftdi1)
target_link_libraries(MyApp PRIVATE PkgConfig::FTDI)

pkg_check_modules(FTDI_CPP REQUIRED IMPORTED_TARGET libftdipp1)
target_link_libraries(MyApp PRIVATE PkgConfig::FTDI_CPP)
```

vcpkg.json
```
{
  "dependencies": ["libftdi1"]
}
```

The include dir in build/vcpkg_installed/x64_linux has libftdi1/ftdi.h but no header for the c++ wrapper and lib also only has libftdi1.a

I have tried to include everything I think is relevant but let me know if more info is needed.

Based on what I understand the files required are missing, but why? I can't do a vcpkg install libftdipp1 as this isn't in the registry. 
I got to the above config by this piece on information:

```
libftdi1 provides pkg-config modules:

  # Library to program and control the FTDI USB controller
  libftdi1

  # C++ wrapper for libftdi1
  libftdipp1
 ```

**Comment by dg0yt** [comment]
AFAICS this port doesn't really build the C++ bindings.

**Comment by dg0yt** [comment]
#40862 enables the C++ wrapper via a feature. Note that this feature is licensed `GPL-2.0-only WITH eCos-exception-2.0`.
